src/libecalc/presentation/yaml/domain/time_series_collection.py

```diff
@@ -0,0 +1,78 @@
+from datetime import datetime
+from typing import List
+
+from typing_extensions import Self, assert_never
+
+from libecalc.common.errors.exceptions import InvalidResource
+from libecalc.dto.types import InterpolationType
+from libecalc.presentation.yaml.domain.time_series import TimeSeries
+from libecalc.presentation.yaml.domain.time_series_exceptions import TimeSeriesNotFound
+from libecalc.presentation.yaml.domain.time_series_provider import TimeSeriesProvider
+from libecalc.presentation.yaml.domain.time_series_resource import TimeSeriesResource
+from libecalc.presentation.yaml.yaml_types.time_series.yaml_time_series import (
+    YamlDefaultTimeSeriesCollection,
+    YamlMiscellaneousTimeSeriesCollection,
+    YamlTimeSeriesCollection,
+)
+
+
+class TimeSeriesCollection(TimeSeriesProvider):
+    """
+    TimeSeriesCollection is a collection of time series (TimeSeriesResource) and common properties for all the time
+    series in the collection.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        resource: TimeSeriesResource,
+        interpolation: InterpolationType,
+        extrapolation: bool,
+        influence_time_vector: bool,
+    ):
+        self.name = name
+        self._resource = resource
+        self._interpolation = interpolation
+        self._extrapolation = extrapolation
+        self._influence_time_vector = influence_time_vector
+
+    def should_influence_time_vector(self) -> bool:
+        return self._influence_time_vector
+
+    def get_time_vector(self) -> List[datetime]:
+        return self._resource.get_time_vector()
+
+    def get_time_series_references(self) -> List[str]:
+        return self._resource.get_headers()
+
+    def get_time_series(self, reference_id: str) -> TimeSeries:
+        try:
+            return TimeSeries(
+                reference_id=f"{self.name};{reference_id}",
```

Comment:
a bit ambigious when both "name+column_name" and column_name is referenced to as reference_id, in different contexts ... can we call it column_name/id here? and that a reference id is "timeseriescollectionname + column name", to get things straight?

Reply:
collection_id and time_series_id? We started using reference_id for resources in the api, in that context it's the name of the resource. 

I agree it's confusing, the way I see it it's just an id, and an id is expected to have meaning only within the context. I.e. time_series.id and collection.id. But it's always useful to then name that id more specifically in other contexts, where you only have the id (time_series_id and collection_id, etc).